### PR TITLE
Document ALTER TABLE RENAME TO limitations

### PR DESCRIPTION
**Pull request by Jessie212**
## Description

Document ALTER TABLE RENAME TO limitations.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 1]
```diff
@@ -373,6 +373,11 @@ statements, the connector supports the following features:
 - {doc}`/sql/show-create-table`
 - {doc}`/sql/comment`
 
+### ALTER TABLE
+
+The connector supports the `ALTER TABLE RENAME TO` command only when used with a
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/delta-lake.md:378]: How about changing "command" to "statement"?

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 2]
```diff
@@ -373,6 +373,11 @@ statements, the connector supports the following features:
...
+Glue metastore.
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/delta-lake.md:379]: We can rename **_external_** tables with Thrift metastore. Can we rephrase this sentence?

Comment (findinpath) [reply docs/src/main/sphinx/connector/delta-lake.md:379]: > The connector supports the `ALTER TABLE RENAME TO`  when the operation does not involve moving the content of the table to a new location on the object storage.

Concrete examples where renaming a table is supported:
- the table type is _external_
- the table is backed by a metastore which does not perform object storage operations, e.g AWS Glue

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 3]
```diff
@@ -373,6 +373,11 @@ statements, the connector supports the following features:
 - {doc}`/sql/show-create-table`
 - {doc}`/sql/comment`
 
+### ALTER TABLE
```
Comment (findinpath) [inline docs/src/main/sphinx/connector/delta-lake.md:376]: Add a `see also` link to  (same as done for Data management)

https://trino.io/docs/current/connector/delta-lake.html#sql-support 

for `ALTER TABLE` option

File: docs/src/main/sphinx/connector/delta-lake.md
Comment (findinpath) [inline docs/src/main/sphinx/connector/delta-lake.md:1]: Use more concise Git commit messages

https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#format-git-commit-messages

**Comment by mosabua** [comment]
Let me know when comments are addressed so I can have a look at an updated version @Jessie212

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 4]
```diff
@@ -367,12 +367,24 @@ statements, the connector supports the following features:
...
-- {doc}`/sql/alter-table`
+- {doc}`/sql/alter-table`, see also {ref}`delta-lake-alter-table`
 - {doc}`/sql/drop-schema`
 - {doc}`/sql/show-create-schema`
 - {doc}`/sql/show-create-table`
 - {doc}`/sql/comment`
 
+
+(delta-lake-alter-table)=
+
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:378]: remove that empty line here

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 5]
```diff
@@ -367,12 +367,24 @@ statements, the connector supports the following features:
 - {doc}`/sql/create-table`, see also {ref}`delta-lake-sql-basic-usage`
 - {doc}`/sql/create-table-as`
 - {doc}`/sql/drop-table`
-- {doc}`/sql/alter-table`
+- {doc}`/sql/alter-table`, see also {ref}`delta-lake-alter-table`
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:370]: This list should be cleaned out and content moved to the delta-lake-data-management section .. or that section overview should be absorbed into here

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 6]
```diff
@@ -367,12 +367,24 @@ statements, the connector supports the following features:
...
 - {doc}`/sql/drop-schema`
 - {doc}`/sql/show-create-schema`
 - {doc}`/sql/show-create-table`
 - {doc}`/sql/comment`
 
+
+(delta-lake-alter-table)=
+
+### ALTER TABLE statement limitations
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:379]: Remove "limitations" from this title and move the whole section to the other ALTER TABLE stuff further below

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 7]
```diff
@@ -367,12 +367,24 @@ statements, the connector supports the following features:
...
+
+The connector only supports the `ALTER TABLE RENAME TO` statement when met with
+the following conditions:
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:382]: this is misleading. .. from what I understand it is one of the the following conditions .. not both ..

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 8]
```diff
@@ -367,12 +367,24 @@ statements, the connector supports the following features:
...
+
+* The table type is external.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:384]: can we link to any section that explains what an external table is?

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 9]
```diff
@@ -367,12 +367,24 @@ statements, the connector supports the following features:
...
+* The table is backed by a metastore that does not perform object storage
+  operations, such as AWS Glue.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:386]: for example AWS Glue

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/connector/delta-lake.md (hunk 8 above)
Comment (Jessie212) [reply docs/src/main/sphinx/connector/delta-lake.md:384]: There's a section in the [Accumulo connector](https://trino.io/docs/current/connector/accumulo.html#external-tables) about external tables. Can I make it more generic, convert it into a fragment, and then place it above the **Table properties** in both connector docs?

Comment (mosabua) [reply docs/src/main/sphinx/connector/delta-lake.md:384]: No.. the accumulo stuff only applies on the high level.. maybe explain in the register table section of the Delta Lake connector (https://trino.io/docs/current/connector/delta-lake.html#register-table) or just above there where it talks about external systems

Comment (Jessie212) [reply docs/src/main/sphinx/connector/delta-lake.md:384]: If other connectors support creating external tables, why is it only mentioned in connection with Accumulo? Documentation on external tables should be a standalone document or section we can link to.

File: docs/src/main/sphinx/connector/delta-lake.md (hunk 5 above)
Comment (Jessie212) [reply docs/src/main/sphinx/connector/delta-lake.md:370]: I think I should do this in a different PR.

File: docs/src/main/sphinx/connector/delta-lake.md [hunk 10]
```diff
@@ -616,6 +616,17 @@ ALTER TABLE test_partitioned_table EXECUTE optimize
 WHERE partition_key = 1
 ```
 
+(delta-lake-alter-table)=
+
+#### ALTER TABLE statement
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.md:621]:
```suggestion
#### ALTER TABLE RENAME TO
```

File: docs/src/main/sphinx/connector/delta-lake.md (hunk 5 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/delta-lake.md:370]: ok

**Review by mosabua — CHANGES_REQUESTED** [review]

**Review by mosabua — APPROVED** [review]

**Pull request merged**